pkg/auth/providers/activedirectory/activedirectory_provider.go

```diff
@@ -228,3 +228,10 @@ func (p *adProvider) getDNAndScopeFromPrincipalID(principalID string) (string, s
 	externalID := strings.TrimPrefix(parts[1], "//")
 	return externalID, scope, nil
 }
+
+func (p *adProvider) GetUserAttributesForAuditLog(token *v3.Token) map[string][]string {
+	extras := make(map[string][]string)
+	extras["principalId"] = []string{token.UserPrincipal.Name}
```

Comment:
I think these need to be lowercase keys based on the description in the code and https://kubernetes.io/docs/reference/access-authn-authz/authentication/#authenticating-proxy
https://github.com/kubernetes/apiserver/blob/master/pkg/authentication/user/user.go#L31